R/zchunk_L241.fgas.R

```diff
@@ -33,80 +32,239 @@ module_emissions_L241.fgas_DISABLED <- function(command, ...) {
+      mutate(USA_factor = if_else(Non.CO2 == "HFC134a", USA_factor/3, USA_factor)) ->
+      L241.hfc_cool_ef_2010_USfactor
+
+    # Format the data frame of 2010 regional emission factors and 2010 USA emission factors
+    # for the next step where future emission factors are calculated.
     #
+    # Future emission factors are will not be calculated for regions with 2010 emission factors
+    # greater than the 2010 USA emission factor because of the way that the calculated as a
+    # fraction of the change between the region and USA 2010 emission factors, negative emission
+    # factors would be estimated.
+    L241.hfc_cool_ef_2010_USfactor %>%
+      filter(USA_factor > value) %>%
+      rename("2010" = `value`, "2030" = `USA_factor`) %>%
+      select(-year) ->
+      L241.hfc_cool_ef_update
+
+    # Estimate future emission factors as the 2010 emission factor plus the difference
+    # the region's and USA's 2010 emission factors multiplied by a scaler for time.
+    L241.hfc_cool_ef_update %>%
+      mutate(`2015` = `2010`+(5 / 20)*(`2030` - `2010`)) %>%
+      mutate(`2020` = `2010`+(10 / 20)*(`2030` - `2010`)) %>%
+      mutate(`2025` = `2010`+(15 / 20)*(`2030` - `2010`)) ->
+      L241.hfc_cool_ef_update_all
+
+    # Subset the future emission factors for the hfc model base years.
     #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # These emission factors will be used in a ratio to compare
+    # future emission factors.
+    L241.hfc_cool_ef_update_all %>%
+      gather(year, value, grep(YEAR_PATTERN, names(.))) %>%
+      filter(!year %in% emissions.HFC_MODEL_BASE_YEARS) ->
+      L241.hfc_cool_ef_update_filtered
+
+
+    # Estimate future emission for non-cooling emissions.
+    #
+    # First, subset the hfc emissions for non-cooling emissions.
+    L141.hfc_R_S_T_Yh  %>%
+      filter(!supplysector %in% c("resid cooling", "comm cooling")) %>%
+      # EF is 1000 x emissions for non-cooling sectors
+      mutate(value = value * 1000) %>%
+      filter(year == max(emissions.HFC_MODEL_BASE_YEARS)) %>%
+      filter(value > 0 ) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      L241.hfc_ef_2010
+
+
+    # Use data from Guus Velders (a f-gas expert) of near future f gas
+    # emissions to calculate the future to 2010 emission factor ratios.
+    # These emission factor ratios will be used to update the non-cooling
+    # emission factors.
+    FUT_EMISS_GV %>%
+      select(-Emissions, -GDP) %>%
+      spread(Year, EF) %>%
+      select(`Species`, `Scenario`, `2010`, `2020`, `2030`) %>%
+      mutate(Species = gsub("-", "", Species )) %>%
```

Comment:
FUT_EMISS_GV and L241.hfc_ef_2010 will be left joined by species. But some of specie names in the FUT_EMISS_GV includes a "-"  where as L241.hfc_ef_2010 species names do not. If you don't remove the "-" some of the species will match up in the left_join in the next step. @bpbond were you asking about the use of gsub? Should I include that in comments?